### PR TITLE
executor: fix two data races in tests

### PR DESCRIPTION
**Pull request by jackysp**
### What problem does this PR solve? 
Fix two data races in tests.

### What is changed and how it works?
Add atomic store for global variables.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

### CODE REVIEW

File: executor/executor_test.go [hunk 1]
```diff
@@ -4097,7 +4109,7 @@ func (s *testSuite) TestOOMPanicAction(c *C) {
 	}
 	tk.Se.SetSessionManager(sm)
 	s.domain.ExpensiveQueryHandle().SetSessionManager(sm)
-	config.GetGlobalConfig().OOMAction = config.OOMActionCancel
+	setOOMAction(config.OOMActionCancel)
```
Comment (coocood) [inline executor/executor_test.go]: Need to set it back?

Comment (jackysp) [reply executor/executor_test.go]: done

**Comment by coocood** [comment]
LGTM

**Comment by jackysp** [comment]
`go test -vet=off -p 5 -timeout 10m -race -count 1 github.com/pingcap/tidb/ddl` reached the test timeout limit and was killed, but I did not change ddl package...

**Pull request merged**